addons/web/static/src/views/list/list_controller.js

```diff
@@ -163,6 +163,24 @@ export class ListController extends Component {
     }
 
     async openRecord(record) {
+        if (this.archInfo.openAction) {
+            this.actionService.doActionButton({
+                name: this.archInfo.openAction.action,
+                type: this.archInfo.openAction.type,
+                resModel: record.resModel,
+                resId: record.resId,
+                resIds: record.resIds,
+                context: record.context,
+                onClose: async () => {
+                    this.defaultOpenRecord(record); //TODO: FIXME? this is a hack in case non-valid action returned
```

Comment:
Why would we want to handle this case? We should reload the record/view instead (when we close the dialog, if the action opens in a dialog)

Reply:
Done. Removed the unnecessary `defaultOpenRecord` as well.